### ISSUE TITLE
Launch DS Browser on DS Lite with Memory Expansion Pak

### ISSUE THREAD

**Issue by getBoolean** [DS-Homebrew/nds-bootstrap#1551]
**Your console:** DS Lite

**Launch Method:**

- Launched using r4 card: Wood version R4ISDHC (r4isdhc.com.cn)
   - [amazon link](https://www.amazon.com/gp/product/B08YJ8R4FS)
- Not sure about SCFG
- I'm using TwilightMenu++ v25-9-1

**Version used:**

TwilightMenu++ v25-9-1, which uses `nds-bootstrap` v0.71.0

**Game tested/used/applicable:**

- Nintendo DS Browser (dumped from cartridge using wooddumper)
- SHA: dc3218e5831d3d34049004fa423856bb540dceca
- MD5: 0c84ea2278ad09ba2dd1b0b745d001dc

**Save file:** [002.NintendoDSBrowser.zip](https://github.com/DS-Homebrew/nds-bootstrap/files/11130629/002.NintendoDSBrowser.zip)

**Expected Behavior:** DS Browser should be able to detect and use the Memory Expansion pak

**Actual Behavior:** DS Browser asks for the memory expansion to be inserted, even though it already is.

![IMG_0622 (Medium)](https://user-images.githubusercontent.com/19920697/229313518-33349401-ead9-4e3d-9474-e9a24994f880.png)

### Steps to reproduce
1. Insert Memory Expansion Pak into Slot 2
2. Turn on DS Lite
3. Launch TwilightMenu++
4. Launch DS Browser rom
5. See above error message in image

### nds-bootstrap.ini

```
[NDS-BOOTSTRAP]
USE_ROM_REGION = 1
DEBUG = 0
LOGGING = 0
MACRO_MODE = 0
SLEEP_MODE = 1
SOUND_FREQ = 0
CONSOLE_MODEL = 0
HOTKEY = 284
NDS_PATH = fat:/apps/002.NintendoDSBrowser.nds
SAV_PATH = fat:/apps/saves/002.NintendoDSBrowser.sav
RAM_DRIVE_PATH = sd:/null.img
GUI_LANGUAGE = en
LANGUAGE = -1
REGION = 1
DSI_MODE = 0
EXTENDED_MEMORY = 0
DONOR_SDK_VER = 0
PATCH_MPU_REGION = 0
PATCH_MPU_SIZE = 0
FORCE_SLEEP_PATCH = 0
```

### Logs

[NDSBTSRP.LOG](https://github.com/DS-Homebrew/nds-bootstrap/files/11130647/NDSBTSRP.LOG)

### Other notes

None

### Will you be available for future issues and/or replies?

Yes

**Comment by spellboundtriangle** [comment]
I can confirm that this happens on a R4 3DS UPGRADE SDHC Revolution for 3DS DSI XL/LL DSL DS GOLD Pro Real Time Save r4isdhc.com 2021 with it working fine on YSMenu

**Comment by PW5190** [comment]
Even though I don't have a MEP myself, I still tried it with my EZ FLASH Omega Definitive Edition flash cartridge, with its Mode B set to RAM. It yielded me the same result shown here. So, I decided to buy a copy of the DS Browser and MEP myself to find out if the outcome is the same for nds-bootstrap.

**Comment by PW5190** [comment]
> Even though I don't have a MEP myself, I still tried it with my EZ FLASH Omega Definitive Edition flash cartridge, with its Mode B set to RAM. It yielded me the same result shown here. So, I decided to buy a copy of the DS Browser and MEP myself to find out if the outcome is the same for nds-bootstrap.

I tried once more, and the issue still persists on TWLMenu++ Release v27.9.0 and nds-bootstrap Release v2.0.1. It would likely may not be resolved in the future.

**Comment by juniornff** [comment]
Same problem Here

**Your console:**
DS Lite

**Flashcard:**
R4i-SDHC RTS 3DS (http://r4i-sdhc.com/)

**Version used:**
TwilightMenu++ v27.12.2, which uses nds-bootstrap v2.2.2

**Game tested/used/applicable:**
Nintendo DS Browser (USA, Europe) (En,Fr,De,Es,It).nds
SHA: dc3218e5831d3d34049004fa423856bb540dceca
MD5: 0c84ea2278ad09ba2dd1b0b745d001dc

**Expected Behavior:**
DS Browser should be able to detect and use the Memory Expansion pak

**Actual Behavior:**
DS Browser asks for the memory expansion to be inserted, even though it already is.

**Steps to reproduce**
Insert Memory Expansion Pak into Slot 2
Turn on DS Lite
Launch TwilightMenu++
Launch DS Browser ROM

**nds-bootstrap.ini**
```
[NDS-BOOTSTRAP]
USE_ROM_REGION = 1
NDS_PATH = fat:/Tools/Nintendo DS Browser (USA, Europe) (En,Fr,De,Es,It).nds
SAV_PATH = fat:/Tools/saves/Nintendo DS Browser (USA, Europe) (En,Fr,De,Es,It).sav
RAM_DRIVE_PATH = sd:/null.img
GUI_LANGUAGE = es
LANGUAGE = 5
REGION = 1
DSI_MODE = 0
CARD_READ_DMA = 1
DONOR_SDK_VER = 0
PATCH_MPU_REGION = 0
PATCH_MPU_SIZE = 0
FORCE_SLEEP_PATCH = 0
DEBUG = 0
LOGGING = 1
MACRO_MODE = 0
SLEEP_MODE = 1
SOUND_FREQ = 0
CONSOLE_MODEL = 0
HOTKEY = 284
AP_FIX_PATH =
BOOST_CPU = 0
BOOST_VRAM = 0
ASYNC_CARD_READ = 0
```

**Additional information**
I already tested the RAM status with this [tool](https://www.gamebrew.org/wiki/MemtestARM) and it works perfectly

When I use the Kernel it works, but the `YSMenu.ini` file changes the value of `AUTO_BOOT` from `BOOT.NDS` to `Nintendo DS Browser (USA, Europe) (En,Fr,De,Es,It).nds` so when I turn the console off and on again I have to use the button combination `(A+B+X+Y+L+R)` to return to TwilightMenu. i have a copy of `YSMenu.ini` so i can set it back, but it's quite annoying

**Comment by TheDocHudson** [comment]
This is known issue, press `X` on the dsbrowser and set launcher to `Kernel.` Kernel _does_ detect the ram expansion

**Comment by juniornff** [comment]
Yes, that is a workaround, but is it known if there are plans to make nds-bootstrap detect and use the MEP?

**Comment by RocketRobz** [comment]
> Yes, that is a workaround, but is it known if there are plans to make nds-bootstrap detect and use the MEP?

I have no idea how to get nds-bootstrap to detect it within games. It does work with the SuperCard MiniSD being used as memory expansion though.

**Comment by juniornff** [comment]
> > Yes, that is a workaround, but is it known if there are plans to make nds-bootstrap detect and use the MEP?
> 
> I have no idea how to get nds-bootstrap to detect it within games. It does work with the SuperCard MiniSD being used as memory expansion though.

I understand, you can't just pull the solution out of nowhere XD.

But as far as I know, the SuperCard Mini SD is used for GB/GBA games, is there a special configuration to make it work as a MEP or the simple fact of having it makes it act as such?

**Comment by RocketRobz** [comment]
> But as far as I know, the SuperCard Mini SD is used for GB/GBA games, is there a special configuration to make it work as a MEP or the simple fact of having it makes it act as such?

Just having it inserted into the console will have TWLMenu++ and nds-bootstrap use it as such.
